### DISCUSSION TITLE
What is the actual downsides to using API routes to download large files?

### DISCUSSION THREAD

**Discussion by IDrumsey** [vercel/next.js#69463] [Help]
### Summary

I'm trying to implement a feature where the user hits a download button and an api route streams all the file data (potentially 1000s of files) to the client side.

The docs say

> API Routes in Next.js are designed to deliver quick responses and are not built to support the transmission of large amounts of data.

See [this](https://nextjs.org/docs/messages/api-routes-response-size-limit#:~:text=API%20Routes%20in%20Next.js%20are%20designed%20to%20deliver%20quick%20responses%20and%20are%20not%20built%20to%20support%20the%20transmission%20of%20large%20amounts%20of%20data.)

Is the only real downside to using api routes for this performance degradation? Or is something critical going to actually break if I do this?

**Comment by Juneezee** [comment]
> potentially 1000s of files

This looks like a good use case of using CDN.

> Or is something critical going to actually break if I do this?

An attacker could easily launch a DoS / DDoS attack to your Next.js server. If your Next.js server is running on a small VM, just a few download requests could easily exhaust it and crash the server
